test/mocks/MockV4Router.sol

```diff
@@ -27,7 +28,11 @@ contract V4RouterImplementation is V4Router, ReentrancyLock {
     }
 
     function _pay(Currency token, address payer, uint256 amount) internal override {
-        ERC20(Currency.unwrap(token)).safeTransferFrom(payer, address(poolManager), amount);
+        if (payer == address(this)) {
+            token.transfer(address(poolManager), amount);
```

Comment:
so I think we both have the same problem in pay where by this point (if its been called from DeltaResolver) it definitely wont be NATIVE currency, but we still check if its native inside currency. I wonder if we should write a currency.transferERC20() without that native check.. anyways food for thought.

Reply:
this contract is actually just a mock for tests - in the universal router (where i implement `_pay`) it doesnt check if its ETH. But its a good thought for posm